### PR TITLE
Python SDK v6 Documentation

### PR DESCRIPTION
**Pull request by mrashed-dev**
# Description
This PR finalizes all the in-code documentation, updates the README.md file and adds an UPGRADE.md file.

# License

I confirm that this contribution is made under the terms of the MIT license and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

**Review by AaronDDM — APPROVED** [review]

File: nylas/resources/calendars.py [hunk 1]
```diff
@@ -21,13 +22,37 @@ class Calendars(
...
     def find(self, identifier: str, calendar_id: str) -> Response[Calendar]:
+        """
+        Return a Calendar
+
+        Args:
+            identifier: The identifier of the grant to act upon.
+            calendar_id: The id of the Calendar to retrieve. Use "primary" to refer to the primary calendar associated with grant.
+
+        Returns:
+            The Calendar.
```
Comment (emarsh1995) [inline nylas/resources/calendars.py:54]:
```suggestion
        Return a Calendar.

        Args:
            identifier: The identifier of the Grant to act upon.
            calendar_id: The ID of the Calendar to retrieve. Use "primary" to refer to the primary Calendar associated with the Grant.

        Returns:
            The Calendar.
```

File: nylas/resources/calendars.py [hunk 2]
```diff
@@ -36,6 +61,16 @@ def find(self, identifier: str, calendar_id: str) -> Response[Calendar]:
     def create(
         self, identifier: str, request_body: CreateCalendarRequest
     ) -> Response[Calendar]:
+        """
+        Create a Calendar
+
+        Args:
+            identifier: The identifier of the grant to act upon.
+            request_body: The values to create the Calendar with.
```
Comment (emarsh1995) [inline nylas/resources/calendars.py:69]:
```suggestion
        Create a Calendar.

        Args:
            identifier: The identifier of the Grant to act upon.
            request_body: The values to create the Calendar with.
```

File: nylas/resources/calendars.py [hunk 3]
```diff
@@ -45,21 +80,43 @@ def create(
     def update(
         self, identifier: str, calendar_id: str, request_body: UpdateCalendarRequest
     ) -> Response[Calendar]:
+        """
+        Update a Calendar
+
+        Args:
+            identifier: The identifier of the grant to act upon.
+            calendar_id: The id of the Calendar to update. Use "primary" to refer to the primary calendar associated with grant.
+            request_body: The values to update the Calendar with.
```
Comment (emarsh1995) [inline nylas/resources/calendars.py:89]:
```suggestion
        Update a Calendar.

        Args:
            identifier: The identifier of the Grant to act upon.
            calendar_id: The ID of the Calendar to update. Use "primary" to refer to the primary Calendar associated with the Grant.
            request_body: The values to update the Calendar with.
```

File: nylas/resources/calendars.py [hunk 4]
```diff
@@ -45,21 +80,43 @@ def create(
...
     def destroy(self, identifier: str, calendar_id: str) -> DeleteResponse:
+        """
+        Delete a Calendar
+
+        Args:
+            identifier: The identifier of the grant to act upon.
+            calendar_id: The id of the Calendar to delete. Use "primary" to refer to the primary calendar associated with grant.
+
+        Returns:
+            The deletion response
```
Comment (emarsh1995) [inline nylas/resources/calendars.py:109]:
```suggestion
        Delete a Calendar.

        Args:
            identifier: The identifier of the Grant to act upon.
            calendar_id: The ID of the Calendar to delete. Use "primary" to refer to the primary Calendar associated with the Grant.

        Returns:
            The deletion response.
```

File: nylas/resources/calendars.py [hunk 5]
```diff
@@ -45,21 +80,43 @@ def create(
...
         return super(Calendars, self).destroy(
             path=f"/v3/grants/{identifier}/calendars/{calendar_id}"
         )
 
     def get_availability(
         self, identifier: str, request_body: GetAvailabilityRequest
     ) -> Response[GetAvailabilityResponse]:
-        """Get availability for a calendar.
+        """
+        Get availability for a calendar.
```
Comment (emarsh1995) [inline nylas/resources/calendars.py:119]:
```suggestion
        Get availability for a Calendar.
```

File: nylas/resources/events.py [hunk 6]
```diff
@@ -35,6 +35,17 @@ class Events(
...
     ) -> ListResponse[Event]:
+        """
+        Return all Events
+
+        Args:
+            identifier: The identifier of the grant to act upon
+            query_params: The query parameters to include in the request
+
+        Returns:
+            The list of Events
```
Comment (emarsh1995) [inline nylas/resources/events.py:46]:
```suggestion
        Return all Events.

        Args:
            identifier: The identifier of the Grant to act upon.
            query_params: The query parameters to include in the request.

        Returns:
            The list of Events.
```

File: nylas/resources/events.py [hunk 7]
```diff
@@ -44,6 +55,18 @@ def list(
...
+        """
+        Return an Event
+
+        Args:
+            identifier: The identifier of the grant to act upon
+            event_id: The id of the Event to retrieve.
+            query_params: The query parameters to include in the request
+
+        Returns:
+            The Event
```
Comment (emarsh1995) [inline nylas/resources/events.py:67]:
```suggestion
        Return an Event.

        Args:
            identifier: The identifier of the Grant to act upon.
            event_id: The ID of the Event to retrieve.
            query_params: The query parameters to include in the request.

        Returns:
            The Event.
```

File: nylas/resources/events.py [hunk 8]
```diff
@@ -56,6 +79,18 @@ def create(
...
+        """
+        Create an Event
+
+        Args:
+            identifier: The identifier of the grant to act upon
+            request_body: The values to create the Event with
+            query_params: The query parameters to include in the request
+
+        Returns:
+            The created Event
```
Comment (emarsh1995) [inline nylas/resources/events.py:91]:
```suggestion
        Create an Event.

        Args:
            identifier: The identifier of the Grant to act upon.
            request_body: The values to create the Event with.
            query_params: The query parameters to include in the request.

        Returns:
            The created Event.
```

File: nylas/resources/events.py [hunk 9]
```diff
@@ -70,6 +105,19 @@ def update(
...
+        Update an Event
+
+        Args:
+            identifier: The identifier of the grant to act upon
+            event_id: The id of the Event to update.
+            request_body: The values to update the Event with
+            query_params: The query parameters to include in the request
+
+        Returns:
+            The updated Event
```
Comment (emarsh1995) [inline nylas/resources/events.py:118]:
```suggestion
        Update an Event.

        Args:
            identifier: The identifier of the Grant to act upon.
            event_id: The ID of the Event to update.
            request_body: The values to update the Event with.
            query_params: The query parameters to include in the request.

        Returns:
            The updated Event.
```

File: nylas/resources/events.py [hunk 10]
```diff
@@ -80,6 +128,18 @@ def update(
...
+        """
+        Delete an Event
+
+        Args:
+            identifier: The identifier of the grant to act upon
+            event_id: The id of the Event to delete.
+            query_params: The query parameters to include in the request
+
+        Returns:
+            The deletion response
```
Comment (emarsh1995) [inline nylas/resources/events.py:140]:
```suggestion
        Delete an Event.

        Args:
            identifier: The identifier of the Grant to act upon.
            event_id: The ID of the Event to delete.
            query_params: The query parameters to include in the request.

        Returns:
            The deletion response.
```

File: nylas/resources/grants.py [hunk 11]
```diff
@@ -22,28 +22,79 @@ class Grants(
     DestroyableApiResource,
 ):
     def list(self, query_params: ListGrantsQueryParams) -> ListResponse[Grant]:
+        """
+        Return all Grants
```
Comment (emarsh1995) [inline nylas/resources/grants.py:26]:
```suggestion
        Return all Grants.
```

File: nylas/resources/grants.py [hunk 12]
```diff
@@ -22,28 +22,79 @@ class Grants(
...
 
     def find(self, grant_id: str) -> Response[Grant]:
+        """
+        Return a Grant
+
+        Args:
+            grant_id: The id of the Grant to retrieve.
+
+        Returns:
+            The Grant
```
Comment (emarsh1995) [inline nylas/resources/grants.py:47]:
```suggestion
        Return a Grant.

        Args:
            grant_id: The ID of the Grant to retrieve.

        Returns:
            The Grant.
```

File: nylas/resources/grants.py [hunk 13]
```diff
@@ -22,28 +22,79 @@ class Grants(
...
 
     def create(self, request_body: CreateGrantRequest) -> Response[Grant]:
+        """
+        Create a Grant
+
+        Args:
+            request_body: The values to create the Grant with.
+
+        Returns:
+            The created Grant
```
Comment (emarsh1995) [inline nylas/resources/grants.py:62]:
```suggestion
        Create a Grant.

        Args:
            request_body: The values to create the Grant with.

        Returns:
            The created Grant.
```

File: nylas/resources/grants.py [hunk 14]
```diff
@@ -22,28 +22,79 @@ class Grants(
...
     ) -> Response[Grant]:
+        """
+        Update a Grant
+
+        Args:
+            grant_id: The id of the Grant to update.
+            request_body: The values to update the Grant with.
+
+        Returns:
+            The updated Grant
```
Comment (emarsh1995) [inline nylas/resources/grants.py:80]:
```suggestion
        Update a Grant.

        Args:
            grant_id: The ID of the Grant to update.
            request_body: The values to update the Grant with.

        Returns:
            The updated Grant.
```

File: nylas/resources/grants.py [hunk 15]
```diff
@@ -22,28 +22,79 @@ class Grants(
...
 
     def destroy(self, grant_id: str) -> DeleteResponse:
+        """
+        Delete a Grant
+
+        Args:
+            grant_id: The id of the Grant to delete.
+
+        Returns:
+            The deletion response
```
Comment (emarsh1995) [inline nylas/resources/grants.py:97]:
```suggestion
        Delete a Grant.

        Args:
            grant_id: The ID of the Grant to delete.

        Returns:
            The deletion response.
```

File: nylas/resources/redirect_uris.py [hunk 16]
```diff
@@ -21,17 +21,44 @@ class RedirectUris(
     DestroyableApiResource,
 ):
     def list(self) -> ListResponse[RedirectUri]:
+        """
+        Return all Redirect URIs
```
Comment (emarsh1995) [inline nylas/resources/redirect_uris.py:25]:
```suggestion
        Return all Redirect URIs.
```

File: nylas/resources/redirect_uris.py [hunk 17]
```diff
@@ -21,17 +21,44 @@ class RedirectUris(
...
 
     def find(self, redirect_uri_id: str) -> Response[RedirectUri]:
+        """
+        Return a Redirect URI
+
+        Args:
+            redirect_uri_id: The id of the Redirect URI to retrieve.
+
+        Returns:
+            The Redirect URI
```
Comment (emarsh1995) [inline nylas/resources/redirect_uris.py:43]:
```suggestion
        Return a Redirect URI.

        Args:
            redirect_uri_id: The ID of the Redirect URI to retrieve.

        Returns:
            The Redirect URI.
```

File: nylas/resources/redirect_uris.py [hunk 18]
```diff
@@ -21,17 +21,44 @@ class RedirectUris(
...
 
     def create(self, request_body: CreateRedirectUriRequest) -> Response[RedirectUri]:
+        """
+        Create a Redirect URI
+
+        Args:
+            request_body: The values to create the Redirect URI with
+
+        Returns:
+            The created Redirect URI
```
Comment (emarsh1995) [inline nylas/resources/redirect_uris.py:59]:
```suggestion
        Create a Redirect URI.

        Args:
            request_body: The values to create the Redirect URI with.

        Returns:
            The created Redirect URI.
```

File: nylas/resources/redirect_uris.py [hunk 19]
```diff
@@ -41,13 +68,34 @@ def create(self, request_body: CreateRedirectUriRequest) -> Response[RedirectUri
...
     ) -> Response[RedirectUri]:
+        """
+        Update a Redirect URI
+
+        Args:
+            redirect_uri_id: The id of the Redirect URI to update.
+            request_body: The values to update the Redirect URI with.
+
+        Returns:
+            The updated Redirect URI
```
Comment (emarsh1995) [inline nylas/resources/redirect_uris.py:79]:
```suggestion
        Update a Redirect URI.

        Args:
            redirect_uri_id: The ID of the Redirect URI to update.
            request_body: The values to update the Redirect URI with.

        Returns:
            The updated Redirect URI.
```

File: nylas/resources/redirect_uris.py [hunk 20]
```diff
@@ -41,13 +68,34 @@ def create(self, request_body: CreateRedirectUriRequest) -> Response[RedirectUri
...
 
     def destroy(self, redirect_uri_id: str) -> DeleteResponse:
+        """
+        Delete a Redirect URI
+
+        Args:
+            redirect_uri_id: The id of the Redirect URI to delete.
+
+        Returns:
+            The deleted Redirect URI
```
Comment (emarsh1995) [inline nylas/resources/redirect_uris.py:96]:
```suggestion
        Delete a Redirect URI.

        Args:
            redirect_uri_id: The ID of the Redirect URI to delete.

        Returns:
            The deletion response.
```

**Review by emarsh1995 — COMMENTED** [review]
Updates to the documentation. Let me know if you have any questions. :)

**Review by emarsh1995 — APPROVED** [review]
LGTM! Thanks so much! 🎉

**Pull request merged**